src/tools/fuzzing.h

```diff
@@ -432,21 +461,83 @@ class TranslateToFuzzReader {
   void setupEvents() {
     Index num = upTo(3);
     for (size_t i = 0; i < num; i++) {
-      auto* event =
-        builder.makeEvent(std::string("event$") + std::to_string(i),
-                          WASM_EVENT_ATTRIBUTE_EXCEPTION,
-                          Signature(getControlFlowType(), Type::none));
-      wasm.addEvent(event);
+      auto name = std::string("event$") + std::to_string(i);
+      if (!wasm.getEventOrNull(name)) {
+        auto* event =
+          builder.makeEvent(name,
+                            WASM_EVENT_ATTRIBUTE_EXCEPTION,
+                            Signature(getControlFlowType(), Type::none));
+        wasm.addEvent(event);
+      }
+    }
+  }
+
+  void finalizeMemory() {
+    for (auto& segment : wasm.memory.segments) {
+      Address maxOffset = segment.data.size();
+      if (!segment.isPassive) {
+        // If the offset is a global that was imported (which is ok) but no
+        // longer is (not ok) we need to change that.
+        if (auto* offset = segment.offset->dynCast<GlobalGet>()) {
+          if (!wasm.getGlobal(offset->name)->imported()) {
+            // TODO: the segments must not overlap...
```

Comment:
Hmm, you're right, it is valid wasm. It would prevent MemoryPacking from running though, so we may want to avoid it anyhow. I'll add that to the comment.